apps/axistream/src/axi_stream_writer.hh

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "hls_macros.hh"
+#include "util.hh"
+#include "axi_stream_common.hh"
+#include "message.hh"
+
+/*
+ * Writes arbitrary length data to an AXI stream.
+ * This adheres to the Xilinx AXI Reference Guide UG 761: https://www.xilinx.com/support/documentation/ip_documentation/ug761_axi_reference_guide.pdf.
+ * For more details, see AxiStreamReader (axi_stream_reader.hh).
+ */
+template <int DataWidth>
+class AxiStreamWriter {
+public:
+    AxiStreamWriter(AxiStream<DataWidth>& stream) : stream_(stream) {
+        HLS_PRAGMA(HLS inline);
+        reset();
```

Comment:
drop-in replacement?

Reply:
Same, only difference are the method names: https://github.com/Xilinx/HLS_packet_processing/blob/master/apps/common/ip.hpp#L3006